girder/utility/plugin_utilities.py

```diff
@@ -17,462 +14,94 @@
-            'name': entry_point.name,
-            'description': '',
-            'version': '',
-            'dependencies': set()
-        }
-        configJson = os.path.join('girder', 'plugin.json')
-        configYml = os.path.join('girder', 'plugin.yml')
-        data = {}
-        try:
-            if pkg_resources.resource_exists(entry_point.name, configJson):
-                with pkg_resources.resource_stream(
-                        entry_point.name, configJson) as conf:
-                    try:
-                        data = json.load(codecs.getreader('utf8')(conf))
-                    except ValueError:
-                        _recordPluginFailureInfo(
-                            plugin=entry_point.name,
-                            traceback=traceback.format_exc())
-                        logprint.exception(
-                            'ERROR: Plugin "%s": plugin.json is not valid '
-                            'JSON.' % entry_point.name)
-            elif pkg_resources.resource_exists(entry_point.name, configYml):
-                with pkg_resources.resource_stream(
-                        entry_point.name, configYml) as conf:
-                    try:
-                        data = yaml.safe_load(conf)
-                    except yaml.YAMLError:
-                        _recordPluginFailureInfo(
-                            plugin=entry_point.name,
-                            traceback=traceback.format_exc())
-                        logprint.exception(
-                            'ERROR: Plugin "%s": plugin.yml is not valid '
-                            'YAML.' % entry_point.name)
-        except (ImportError, SystemError):
-            # Fall through and just try to load the entry point below.  If
-            # there is still an error, we'll log it there.
-            pass
-        if data == {}:
-            try:
-                data = getattr(entry_point.load(), 'config', {})
-            except (ImportError, SystemError):
-                # If the plugin failed to load via entrypoint, but is in the
-                # plugins directory, it may still load.  We mark and report the
-                # failure; if it loads later, the failure will be cleared, but
-                # the report is still desired.
-                _recordPluginFailureInfo(
-                    plugin=entry_point.name, traceback=traceback.format_exc())
-                logprint.exception(
-                    'ERROR: Plugin "%s": could not be loaded by entrypoint.' % entry_point.name)
-                continue
-        allPlugins[entry_point.name].update(data)
-        allPlugins[entry_point.name]['dependencies'] = set(
-            allPlugins[entry_point.name]['dependencies'])
+    mgr = NamedExtensionManager(
+        namespace=NAMESPACE,
+        names=plugins,
+        invoke_on_load=True,
+        invoke_args=(info,),
+        warn_on_missing_entrypoint=True,
+        verify_requirements=True
+    )
+    return mgr.names()
```

Comment:
This uses `stevedore` to initialize all plugins from a list.  All that it does is loop through the names, find the associated entrypoint, and calls it with the given arguments.  Again, there are hooks we can add for error handling.